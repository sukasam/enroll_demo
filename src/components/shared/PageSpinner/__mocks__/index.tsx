export default function PageSpinner(): JSX.Element {
    return <div data-testid="page-spinner">Loading...</div>;
}
