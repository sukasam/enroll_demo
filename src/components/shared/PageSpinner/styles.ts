import { css } from "@emotion/react";

export default css`
    position: fixed;
    top: 0;
    left: 0;
    width: 100%;
    height: 100%;
    background-color: rgba(255, 255, 255, 0.5); // Changed to transparent white
    display: flex;
    justify-content: center;
    align-items: center;
    z-index: 9999;
`;
