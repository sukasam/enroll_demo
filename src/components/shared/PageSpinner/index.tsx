/** @jsxImportSource @emotion/react */
import Spinner from "Components/shared/Spinner";
import styles from "./styles";

function PageSpinner(): JSX.Element {
    return (
        <div css={styles}>
            <Spinner height="100px" data-testid="page_spinner" dark />
        </div>
    );
}

export default PageSpinner;
