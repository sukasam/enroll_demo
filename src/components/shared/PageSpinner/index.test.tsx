import { render } from "@testing-library/react";
import PageSpinner from "./index";

jest.mock(
    "Components/shared/Spinner",
    () =>
        function MockSpinner(props: any): JSX.Element {
            return <div data-testid="mock-spinner" {...props} />;
        }
);

// Mock the styles
jest.mock("./styles", () => ({
    __esModule: true,
    default: {
        display: "flex",
        justifyContent: "center",
        alignItems: "center"
    }
}));

describe("PageSpinner", () => {
    it("matches snapshot", (): void => {
        const { container } = render(<PageSpinner />);
        expect(container).toMatchSnapshot();
    });
});
