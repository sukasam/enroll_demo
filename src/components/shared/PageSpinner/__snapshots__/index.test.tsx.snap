// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PageSpinner matches snapshot 1`] = `
<div>
  <div
    class="css-1n2mv2k"
  >
    <div
      data-testid="page_spinner"
      height="100px"
    />
  </div>
</div>
`;
